pilot/cmd/pilot-agent/status/server.go

```diff
@@ -189,8 +192,16 @@ func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
 	httpClient := &http.Client{
 		// TODO: figure out the appropriate timeout?
 		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	var url string
+	if prober.Scheme == corev1.URISchemeHTTPS {
+		url = fmt.Sprintf("https://127.0.0.1:%v%s", prober.Port.IntValue(), prober.Path)
+	} else {
```

Comment:
This if/else assume either HTTPS or HTTP, which is true today.  I prefer we to put condition explicitly, `prober.Scheme == corev1.URISchemeHTTP`

Reply:
`Scheme ` can be HTTP/ HTTPS /"" default to HTTP,  which is validated by kubernetes

Reply:
Optional, you can use `fmt.Sprintf("%s://127.0.0.1:%v%s", toLowerCase(prober.Scheme), port, path)`.

Reply:
But prober.Scheme` can be "" for the tests.